onnxruntime/core/optimizer/matmul_bn_fusion.cc

```diff
@@ -0,0 +1,280 @@
+#include "core/optimizer/matmul_bn_fusion.h"
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/utils.h"
+
+
+namespace onnxruntime
+{
+void AddNodesToRemove(Node::NodeConstIterator current_iterator,
+                      const NodeIndex& dest_node_index,
+                      std::vector<NodeIndex>& nodes_to_remove) {
+  while (current_iterator->Index() != dest_node_index) {
+    nodes_to_remove.push_back(current_iterator->Index());
+    current_iterator = current_iterator->OutputNodesBegin();
+  }
+}
+
+NodeIndex GetOtherParentOfNode(const Node& node, NodeIndex first_parent_index) {
+  NodeIndex other_parent_index = std::numeric_limits<size_t>::max();
```

Comment:
Can we return an `std::optional` instead of this magic value? It's more intuitive to use and forces the caller to actually check that the value is not null.